src/ooxml/java/org/apache/poi/xssf/usermodel/XSSFCell.java

```diff
@@ -1311,4 +1312,22 @@ void notifyArrayFormulaChanging(){
                 "You cannot change part of an array.";
         notifyArrayFormulaChanging(msg);
     }
+    
+    
+    //Moved from XSSFRow.shift(). Not sure what is purpose. 
```

Comment:
`protected void XSSFRow#shift(int n)` is called by XSSFSheet#shiftRows.
The code does 3 things as part of a row shift

1. Update array formulas (advanced feature; don't worry about this for column shifting yet)
2. Update the calculation chain (advanced feature; likewise)
3. Updates every CTCell's address due to a row shift via `ctCell.setR(r);`
In sheet#.xml, this looks like `<c r="A1" t="n" v="15" />`, where c means cell, r means reference, v means value, and t means type. Failing to update setR would mean that the cell would never get moved.

Are `getRowIndex()` and `getColumnIndex()` updated to the new row and column indices by the time this method is called? It looks like `XSSFRow#shift` was responsible for updating this.
My guess why this code wasn't moved into XSSFCell was because of the number of variables that would have to be passed in as arguments or recomputed by XSSFCell (this is a tight loop: recomputing sheet index and calculation chain for every cell in the shift range is prohibitively expensive).

If you have your development environment set up to run unit tests, you probably would have seen your change fail the row shifting unit tests, which would give some insight into what this code is accomplishing.

XSSFRow.java:
```
protected void shift(int n) {
    int rownum = getRowNum() + n;
    for (Cell c : this) {
        ...
        String r = new CellReference(rownum, cell.getColumnIndex()).formatAsString();
        ...
    }
    setRowNum(rownum);
}
```